keps/sig-storage/4958-csi-sidecars-all-in-one/README.md

```diff
@@ -0,0 +1,1141 @@
+
+
+
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+![overview](./aio6.png "overview of design details")
+
+#### Glossary
+- Individual repository - An existing repository in the kubernetes-csi/ org in Github e.g. the external-attacher repository.
+- AIO monorepo or monorepo - The monolithic repository where most of the code of the CSI Sidecars will be migrated.
+- Monorepo component - The source code of an individual repository that is currently being migrated or already migrated to the monorepo. 
+
+#### AIO Monorepo 
+
+##### Release Management
+
+Semantic Versioning
+
+##### RBAC policy
+
+We designed the AIO repo's RBAC policy to mirror that of individual repos, where each controller maintains its own policy. Driver maintainers should apply proper RBAC when enabling specific controllers in AIO
+
+> We plan to combine informer caches of different controllers in the future, which will involve updating some RBAC policies, to be addressed in a subsequent KEP
+
+##### Command Line
+
+Divided the command lines into two types, a generic command line whose configuration is common to all controllers and is configured only once, and the other type of command lines whose configuration is different for each controller. these command lines each has a new unique name. prefix with the controller name.
+
+example PR: https://github.com/kubernetes-csi/external-attacher/pull/620
+
+#### Monorepo component
+
+poc version: https://github.com/mauriciopoppe/csi-sidecars
```

Comment:
This should be a link to the nested locations in the monorepo source code e.g. https://github.com/mauriciopoppe/csi-sidecars/tree/main/pkg/attacher